lib/cocoapods/installer.rb

```diff
@@ -279,8 +279,18 @@ def validate_build_configurations
     #
     def clean_sandbox
       sandbox.public_headers.implode!
```

Comment:
@segiddins I may take that back. Headers might cause recompilation of swift modules...objective-c does not seem to be re-running. I will spend today/tomorrow to finalize it if I need to do more changes.

*still* this remains a big improvement as is.